envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -31,40 +32,75 @@ class SnapshotUpdater(
         properties = properties
     )
 
-    fun start(changes: Flux<List<LocalityAwareServicesState>>): Flux<List<LocalityAwareServicesState>> {
-        // see GroupChangeWatcher
-        return Flux.combineLatest(
-            onGroupAdded,
-            changes,
-            BiFunction<Any, List<LocalityAwareServicesState>, List<LocalityAwareServicesState>> { _, states -> states }
-        )
-            .sample(properties.stateSampleDuration)
-            .publishOn(scheduler)
-            .doOnNext { states ->
-                versions.retainGroups(cache.groups())
-                updateSnapshots(states)
+    fun start(changes: Flux<List<LocalityAwareServicesState>>): Flux<UpdateResult> {
+        return Flux.merge(
```

Comment:
Do `merge` inherit scheduler from its sources? 
If we don't know, we can simply test it by adding a log line and print current thread name in `doOnNext`

If not, maybe we should add `subscribeOn` after `merge`.

Reply:
It seems they do.